services/metadactyl-clj/src/metadactyl/core.clj

```diff
@@ -63,15 +63,16 @@
 
 (defn -main
   [& args]
-  (require 'metadactyl.routes.api)
-  (let [app (eval 'metadactyl.routes.api/app)
-        {:keys [options arguments errors summary]} (ccli/handle-args config/svc-info args cli-options)]
-    (when-not (fs/exists? (:config options))
-      (ccli/exit 1 (str "The config file does not exist.")))
-    (when-not (fs/readable? (:config options))
-      (ccli/exit 1 "The config file is not readable."))
-    (load-config-from-file (:config options))
-    (tasks/set-logging-context! config/svc-info)
-    (tasks/schedule-tasks)
-    (log/warn "Listening on" (config/listen-port))
-    (jetty/run-jetty app {:port (config/listen-port)})))
+  (tc/with-logging-context config/svc-info
+    (require 'metadactyl.routes.api)
+    (let [app (eval 'metadactyl.routes.api/app)
+          {:keys [options arguments errors summary]} (ccli/handle-args config/svc-info args cli-options)]
+      (when-not (fs/exists? (:config options))
+        (ccli/exit 1 (str "The config file does not exist.")))
+      (when-not (fs/readable? (:config options))
+        (ccli/exit 1 "The config file is not readable."))
+      (load-config-from-file (:config options))
+      (tasks/set-logging-context! config/svc-info)
```

Comment:
I'm not referring to the call to `with-logging-context`. I mean the `tasks/set-logging-context!` call, which changes a ref to include that data, which is later used in another `with-logging-context` call. What I'm not sure about is if that's necessary, or if it's already inheriting that context from the `with-logging-context` here on line 66, such that all that handling with refs and such can be removed. I looked at it briefly when adding `with-logging-context` but I didn't want to get too mired down in testing this periodic-task system so I moved on quickly once it was at least not using `clear-context!`, which I wanted to avoid since it breaks the nested logging context paradigm.

Mentioned it here because you're removing other duplicate context-setting (middlewares vs. what's here), and this might meet the same description.